docs/blog/2022-10-19-text4shell-cve-2022-42889.mdx

```diff
@@ -0,0 +1,156 @@
+  ~
+  ~ https://creativecommons.org/licenses/by-sa/4.0/legalcode
+  ~
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  ~
+-->
+
+## What is Text4Shell?
+
+Text4Shell is a vulnerability in the Java library Apache Commons Text. This vulnerability, in specific conditions,
+allows an attacker to execute arbitrary code on the victim's machine (Remote Code Execution or "RCE").
+The vulnerability was discovered by [Alvaro Muñoz (aka pwntester)](https://twitter.com/pwntester) and
+[announced publicly](https://lists.apache.org/thread/n2bd4vdsgkqh2tm14l1wyc3jyol7s1om) on October 13th
+
+Text4Shell was officially assigned the [CVE-2022-42889](https://nvd.nist.gov/vuln/detail/CVE-2022-42889) identifier.
+
+<!--truncate-->
+
+Note: While the vulnerability is also known as "Act4Shell", we'll be referring to it as "Text4Shell" in this article.
+
+## Who is impacted?
+
+Applications that use the Apache Commons Text library (or that depend on libraries using it) are impacted.
+
+While the name Text4Shell may suggest that this issue is similar to the
+[Log4Shell vulnerability](https://lunasec.io/docs/blog/log4shell-cve-2021-44228/), that is fortunately not the case.
+
+Text4Shell is fortunately similar to [Spring4Shell](https://www.lunasec.io/docs/blog/spring-rce-vulnerabilities/) in
+that exploitable requires a very specific set of conditions for it to be exploitable.
+
+We'll go into more detail about the conditions in the next section, but we believe that the impact of this vulnerability
+is quite limited and that it is unlikely that many applications are impacted. If you'd like to check if your application
+is impacted, you can [scan your code with LunaTrace](https://lunatrace.lunasec.io/) for free. We'll be providing
+additional resources at the bottom of this article as well.
+
+### Exploit Conditions
+
+- Apache Commons Text version 1.5 through 1.9 are vulnerable
+    ([version 1.10](https://commons.apache.org/proper/commons-text/changes-report.html#a1.10.0) includes a fix),
+- The application must use the `StringSubstitutor` interpolator in a vulnerable configuration (see POC code below),
+- The application must accept arbitrary, attacker controlled input for exploitation to be possible,
+- All versions of the JDK are vulnerable (early reports suggest JDK versions 15+ are safe, but new POCs have been
+    released that work on JDK 15+).
+
+### Proof-of-Concept Code
+
+There are a few excellent resources available for this vulnerability already, including the original
+[GitHub Advisory](https://securitylab.github.com/advisories/GHSL-2022-018_Apache_Commons_Text/) and a
+[Docker POC on GitHub](https://github.com/karthikuj/cve-2022-42889-text4shell-docker), and this post will just be
+consolidating the information in those resources together.
+
+The following code snippet is a proof-of-concept that demonstrates how the vulnerability can be exploited.
+
+```typescript
```

Comment:
this is Java not ts?

Reply:
syntax highlighting in docusaurus doesn't work for `java` for some reason. This was my quick hack for Log4Shell lol